build.sc

```diff
@@ -51,11 +51,12 @@ trait SafeDeps extends ScalaModule {
 }
 
 trait MiMaChecks extends Mima {
-  def mimaPreviousVersions = Seq("0.9.0", "0.9.1", "0.9.2", "0.9.3", "0.10.0")
+  def mimaPreviousVersions = Seq("0.9.0", "0.9.1", "0.9.2", "0.9.3", "0.10.0", "0.10.1", "0.10.2", "0.10.3", "0.10.4")
   override def mimaBinaryIssueFilters: T[Seq[ProblemFilter]] = Seq(
     ProblemFilter.exclude[ReversedMissingMethodProblem]("os.PathConvertible.isCustomFs"),
     // this is fine, because ProcessLike is sealed (and its subclasses should be final)
-    ProblemFilter.exclude[ReversedMissingMethodProblem]("os.ProcessLike.joinPumperThreadsHook")
+    ProblemFilter.exclude[ReversedMissingMethodProblem]("os.ProcessLike.joinPumperThreadsHook"),
+    ProblemFilter.exclude[MissingTypesProblem]("os.proc$")
```

Comment:
I am not sure if this is a real problem.
I tried in many ways to add this but I couldn't.
The way the companion object is generated when you have a case class with a vararg argument seems not reproducible, by writing the type by hand. For example: what is the type of a Vararg? 🤔 `Seq[Shellable]` while correct for binary compatibility is not the same at compile time.

Reply:
Could you post the output of MiMa?

Reply:
```
Found 3 issue when checking against com.lihaoyi:os-lib_2.12:0.10.4
 * static method andThen(scala.Function1)scala.Function1 in class os.proc does not have a correspondent in current version
   filter with: ProblemFilter.exclude[DirectMissingMethodProblem]("os.proc.andThen")
 * static method compose(scala.Function1)scala.Function1 in class os.proc does not have a correspondent in current version
   filter with: ProblemFilter.exclude[DirectMissingMethodProblem]("os.proc.compose")
 * the type hierarchy of object os.proc is different in current version. Missing types {scala.runtime.AbstractFunction1}
   filter with: ProblemFilter.exclude[MissingTypesProblem]("os.proc$")
```

Reply:
e.g. we can put it on os.Subprocess instead, since now we os.call and os.spawn not everything goes through os.proc anymore